### DISCUSSION TITLE
Outputting multiple flextable tables in an if

### DISCUSSION THREAD

**Discussion by sunliangxing** [davidgohel/flextable#609] [Q&A] [answered]
Hello，
I am using Rmarkdown to generate word reports. When I use the if judgment statement, I find that the flextable only works on the last line, i.e. the output to word will only show the last table. When I don't use if it outputs the two tables properly.Here is my code
```{r}
a<-as.data.frame(matrix(rnorm(25) , nrow = 5))
colnames(a)<-LETTERS[1:5]
rownames(a)<-LETTERS[6:10]
if(nrow(a)>1){
  flextable(a[1:2,1:2])
  flextable(a[3:5,3:5])
}
```
![image](https://github.com/davidgohel/flextable/assets/71911098/19b8e1e5-6d74-425c-a899-ac5b5ee65d36)

**Comment by davidgohel** [comment, marked as answer]
There are two options for now:

- Using `knit_child()`, this was the recommandation of Posit: https://ardata-fr.github.io/flextable-book/rendering.html#looping-in-r-mardown-documents
- Repeat the `if`twice.

I can try to have something easier automating the usage of `knit_child()`. I'll let you know soon
